File: apps/ocr-worker/src/mime_type.py
"""Portage Python de packages/utils/src/mimeType.ts (`fileTypeFromMimeType`), copie identique de
apps/thumbnail-worker/src/mime_type.py.
"""

FileTypeCategory = str  # "pdf" | "image" | "video" | "docx" | "pptx" | "markdown" | "other"

_DOCX_MIME_TYPES = {
    "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
    "application/msword",
}
_PPTX_MIME_TYPES = {
    "application/vnd.openxmlformats-officedocument.presentationml.presentation",
    "application/vnd.ms-powerpoint",
}
_MARKDOWN_MIME_TYPES = {"text/markdown", "text/x-markdown"}


def file_type_from_mime_type(mime_type: str) -> FileTypeCategory:
    if mime_type == "application/pdf":
        return "pdf"
    if mime_type.startswith("image/"):
        return "image"
    if mime_type.startswith("video/"):
        return "video"
    if mime_type in _MARKDOWN_MIME_TYPES:
        return "markdown"
    if mime_type in _DOCX_MIME_TYPES:
        return "docx"
    if mime_type in _PPTX_MIME_TYPES:
        return "pptx"
    return "other"

File: packages/utils/src/mimeType.ts
import type { FileTypeCategory } from "@docysen/types";

/**
 * Dérive la facette de filtre/l'icône de fallback à partir du `mimeType` stocké en base. Partagé
 * entre l'indexation Meilisearch (api-service) et le fallback icône (frontend) pour ne pas avoir
 * deux logiques de classification qui divergent.
 */
export function fileTypeFromMimeType(mimeType: string): FileTypeCategory {
  if (mimeType === "application/pdf") return "pdf";
  if (mimeType.startsWith("image/")) return "image";
  if (mimeType.startsWith("video/")) return "video";
  if (mimeType === "text/markdown" || mimeType === "text/x-markdown") return "markdown";
  if (
    mimeType === "application/vnd.openxmlformats-officedocument.wordprocessingml.document" ||
    mimeType === "application/msword"
  ) {
    return "docx";
  }
  if (
    mimeType === "application/vnd.openxmlformats-officedocument.presentationml.presentation" ||
    mimeType === "application/vnd.ms-powerpoint"
  ) {
    return "pptx";
  }
  return "other";
}
